src/main/java/com/github/bibsysdev/nva/cristin/person/CristinApiClient.java

```diff
@@ -0,0 +1,190 @@
+import nva.commons.core.JsonUtils;
+import nva.commons.core.attempt.Failure;
+import org.apache.http.client.utils.URIBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@JacocoGenerated
+public class CristinApiClient {
+
+    protected static final String PERSON_LOOKUP_CONTEXT_URL = "https://example.org/person-context.json";
+    private static final Logger logger = LoggerFactory.getLogger(CristinApiClient.class);
+    private static final Environment ENVIRONMENT = new Environment();
+    protected static final String BASE_URL = ENVIRONMENT.readEnv("BASE_URL");
+    private static final String CRISTIN_API_HOST = ENVIRONMENT.readEnv("CRISTIN_API_HOST");
+    private static final ObjectMapper OBJECT_MAPPER = JsonUtils.objectMapper;
+
+    private static final String NAME = "name";
+    private static final String CHARACTER_EQUALS = "=";
+    private static final String HTTPS = "https";
+    private static final String CRISTIN_API_PERSONS_PATH = "/v2/persons/";
+    private static final String SEARCH_PATH = "search?QUERY_PARAMS"; // TODO: NP-2412: Replace QUERY_PARAMS
+    private static final String ERROR_MESSAGE_FETCHING_CRISTIN_PERSON_WITH_ID = "Error fetching cristin person with "
+        + "id: %s . Exception Message: %s";
+    private static final String ERROR_MESSAGE_BACKEND_FETCH_FAILED = "Your request cannot be processed at this time "
+        + "due to an upstream error";
+
+    /**
+     * Creates a wrapper object containing CristinPersons transformed to NvaPersons. Is used for serialization to the
+     * client.
+     *
+     * @param parameters The query params
+     * @param language   Language used for some properties in Cristin API response
+     * @return a PersonsWrapper with transformed CristinPersons
+     * @throws IOException        for connection errors
+     * @throws URISyntaxException if URI is malformed
+     */
+    public PersonsWrapper queryCristinPersonsIntoWrapperObject(Map<String, String> parameters, String language)
+        throws IOException, URISyntaxException {
+
+        long startRequestTime = System.currentTimeMillis();
+        List<CristinPerson> enrichedPersons = queryAndEnrichPersons(parameters, language);
+        long endRequestTime = System.currentTimeMillis();
+
+        PersonsWrapper personsWrapper = new PersonsWrapper();
+        personsWrapper.setId(buildUri(BASE_URL, SEARCH_PATH));
+        personsWrapper.setSize(0); // TODO: NP-2385: X-Total-Count header from Cristin response
+        personsWrapper.setSearchString(extractNameSearchString(parameters));
+        personsWrapper.setProcessingTime(calculateProcessingTime(startRequestTime, endRequestTime));
+        // TODO: NP-2385: Use Link header / Pagination data from Cristin response in the next two values
+        personsWrapper.setFirstRecord(0);
+        personsWrapper.setNextResults(null); // TODO: Change to URI
+        personsWrapper.setHits(transformCristinPersonsToNvaPersons(enrichedPersons));
```

Comment:
Refactor to method

Reply:
to what method? if queryCristinPersonsIntoWrapperObject is not a descriptive name for this content we should refactor the name not make another method imo